packages/liquid-html-parser/src/stage-1-cst.ts

```diff
@@ -110,11 +111,18 @@ export interface ConcreteBasicNode<T> {
 export interface ConcreteLiquidDocParamNode
   extends ConcreteBasicNode<ConcreteNodeTypes.LiquidDocParamNode> {
   name: 'param';
-  paramName: ConcreteTextNode;
+  paramName: ConcreteLiquidDocParamNameNode;
   paramDescription: ConcreteTextNode | null;
   paramType: ConcreteTextNode | null;
 }
 
+export interface ConcreteLiquidDocParamNameNode
```

Comment:
The syntax for default value and required properties are related to how the `ParamName` is defined, so I'm choosing to include that information in this `ConcreteLiquidDocParamNameNode` 

This makes it a bit funky to access, but this aligns with how the parsing logic works, and we map this to a more sensible location in `stage-2`